src/Type/TypeCombinator.php

```diff
@@ -696,16 +696,18 @@ private static function reduceArrays(array $constantArrays): array
 			$arraysToProcess[] = $constantArray;
 		}
 
-		for ($i = 0; $i < count($arraysToProcess); $i++) {
-			for ($j = $i + 1; $j < count($arraysToProcess); $j++) {
+		for ($i = 0, $arraysToProcessCount = count($arraysToProcess); $i < $arraysToProcessCount; $i++) {
+			for ($j = $i + 1; $j < $arraysToProcessCount; $j++) {
 				if ($arraysToProcess[$j]->isKeysSupersetOf($arraysToProcess[$i])) {
 					$arraysToProcess[$j] = $arraysToProcess[$j]->mergeWith($arraysToProcess[$i]);
 					array_splice($arraysToProcess, $i--, 1);
+					$arraysToProcessCount--;
```

Comment:
this does change the loop limit boundary, if it desired, why is that correct, if not, I belive this can not improve performance

Reply:
Although the complexity of count() is O(1), storing the result of count() in a variable is faster than calling them again.

Reply:
But it is really faster than? Please provide microbenchmark which shows `count(array)` is really slower.

Reply:
the problem is not with the complexity O(1) of `count()`, the point is that referencing to where the array's length is stored tends to be slower than accessing data from a variable. This makes them meaningful in the loop.

https://3v4l.org/VCvdQ

Reply:
yes, it seems a little bit faster https://3v4l.org/i2dqh